app/scripts/modules/core/src/pipeline/config/actions/templateJson/ShowPipelineTemplateJsonModal.tsx

```diff
@@ -0,0 +1,108 @@
+
+export interface IShowPipelineTemplateJsonModalProps extends IModalComponentProps {
+  ownerEmail: string;
+  pipeline: IPipeline;
+}
+
+export interface IShowPipelineTemplateJsonModalState {
+  template: IPipelineTemplateV2;
+}
+
+export class ShowPipelineTemplateJsonModal extends React.Component<
+  IShowPipelineTemplateJsonModalProps,
+  IShowPipelineTemplateJsonModalState
+> {
+  public static defaultProps: Partial<IShowPipelineTemplateJsonModalProps> = {
+    dismissModal: noop,
+  };
+
+  constructor(props: IShowPipelineTemplateJsonModalProps) {
+    super(props);
+
+    const template = PipelineTemplateV2Service.createPipelineTemplate(cloneDeep(props.pipeline), props.ownerEmail);
+    this.state = { template };
+  }
+
+  private onChange = (e: React.ChangeEvent<HTMLInputElement>, property: string) =>
+    this.setState({
+      template: {
+        ...this.state.template,
+        metadata: {
+          ...this.state.template.metadata,
+          [property]: e.target.value,
+        },
+      },
+    });
+
+  public render() {
+    const { dismissModal } = this.props;
+    const { template } = this.state;
+    const templateStr = JsonUtils.makeSortedStringFromObject(template);
+
+    return (
+      <div className="flex-fill">
+        <div className="modal-header">
+          <h3>Pipeline Template JSON</h3>
+        </div>
+        <div className="modal-body flex-fill">
+          <p>The JSON below represents the derived template from the pipeline configuration in its persisted state.</p>
+          <form className="pipeline-template-form">
+            <h4>Edit Template</h4>
+            <div className="form-group">
+              <label htmlFor="template-name">Name</label>
+              <input
+                id="template-name"
+                className="form-control input-sm"
+                type="text"
+                value={template.metadata.name}
+                onChange={e => this.onChange(e, 'name')}
+              />
+            </div>
+            <div className="form-group">
+              <label htmlFor="template-description">Description</label>
+              <input
+                id="template-description"
+                className="form-control input-sm"
+                type="text"
+                value={template.metadata.description}
+                onChange={e => this.onChange(e, 'description')}
+                placeholder="Template Description"
+              />
+            </div>
+            <div className="form-group">
+              <label htmlFor="template-owner">Owner</label>
+              <input
+                id="template-owner"
+                className="form-control input-sm"
+                type="text"
+                value={template.metadata.owner}
+                onChange={e => this.onChange(e, 'owner')}
+              />
+            </div>
+            <div className="form-group">
+              <label className="pipeline-template-form__copy">
+                {commandCopy}
```

Comment:
Since the `toolTip` on the `CopyToClipboard` component already contains this text I'm tempted not to include it here. Another thing that I've seen is that the copy to clipboard button tends to be pushed over to the right side of the component it's copying from.  Couple of examples:

![screen shot 2019-02-25 at 2 16 23 pm](https://user-images.githubusercontent.com/34253460/53362415-1668cf80-3908-11e9-8b2c-f3a2ff384ccf.png)

![screen shot 2019-02-25 at 2 16 12 pm](https://user-images.githubusercontent.com/34253460/53362424-1a94ed00-3908-11e9-8100-d693c1a6ca8a.png)

I suggest pushing this icon over to the right as well as reducing the margin between the icon and the JsonEditor component to better indicate that the copy button is related to it.

Reply:
thanks for these great examples. So are you thinking the json editor would be slightly reduced in width and the copy button would sit on the lower right of it?

Reply:
Was thinking something like this:

<img width="330" alt="screen shot 2019-02-27 at 10 15 55 am" src="https://user-images.githubusercontent.com/34253460/53500639-c2c9c380-3a78-11e9-9403-74956aac0063.png">

Where the green box is the copy-to-clipboard icon

Reply:
Awesome, thanks